src/Map.ts

```diff
@@ -595,13 +600,47 @@ export class Map extends maplibregl.Map {
       const strMoreInfoRegex = /^(.*)({\s*name\s*(:\s*(\S*))?\s*})(.*)$/;
 
       const langStr = language ? `name:${language}` : "name"; // to handle local lang
-      const replacer = [
+      let replacer = [
         "case",
         ["has", langStr],
         ["get", langStr],
         ["get", "name"],
       ];
 
```

Comment:
The replaceer section can be replaced with `ExpressionSpecification` (a type from maplibre-gl):

```ts
let replacer: ExpressionSpecification = [
        "case",
        ["has", langStr],
        ["get", langStr],
        ["get", "name"],
      ];

      if (language == Language.VISITOR) {
        const firstLang = `name:${getBrowserLanguage()}`;
        replacer = [
          "case",
          ["all", ["has", firstLang], ["has", "name"]],

          [
            "case",
            ["==", ["get", firstLang], ["get", "name"]],
            ["get", "name"],
            ["concat", ["get", firstLang], "\n", ["get", "name"]],
          ],
          ["get", "name"],
        ];
      }
      ...
```

then you can avoid all the `@ts-ignore`